src/Routing/QueryRouteEnhancer.php

```diff
@@ -98,22 +116,23 @@ protected function enhanceBatch($defaults, $values) {
    */
   protected function enhanceSingle($defaults, $values) {
     $values = array_intersect_key($values + [
-      'query' => '',
-      'variables' => [],
-      'id' => '',
-    ], array_flip(['query', 'variables', 'id']));
-
-    $query = $values['query'];
-    $id = $values['id'];
-    $variables = $values['variables'];
-    if (empty($query) && empty($id)) {
+      'query' => NULL,
+      'variables' => NULL,
+      'id' => NULL,
+      'version' => NULL,
+    ], array_flip(['query', 'variables', 'id', 'version']));
+
+    $variables = $this->getVariables($values['variables']);
+    if (empty( $values['query']) && empty($values['id'])) {
+      return FALSE;
+    }
+
+    if (empty($values['query']) && empty($values['id']) && empty($values['version'])) {
       return FALSE;
     }
```

Comment:
I believe you didn't resolve the conflict in this method properly. I changed the code here a bit and you brought back some of the things I removed. Can you check the latest code in your merge target?